tags/tags-impl/impl/src/java/org/sakaiproject/tags/impl/job/xml/Value.java

```diff
@@ -0,0 +1,402 @@
+
+/**
+ * Copyright (c) 2012 scireum GmbH - Andreas Haufler - aha@scireum.de
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package org.sakaiproject.tags.impl.job.xml;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.util.regex.Pattern;
+/**
+ * Provides a simple wrapper for a value which is read from an untyped context
+ * like HTTP parameters. It supports elegant <code>null</code> handling and type
+ * conversions.
+ */
+public class Value {
+    private Object data;
+
+    /**
+     * Determines if the wrapped value is null.
+     */
+    public boolean isNull() {
+        return data == null;
+    }
+
+    /**
+     * Determines if the wrapped value is an empty string.
+     */
+    public boolean isEmptyString() {
+        return data == null || "".equals(data);
```

Comment:
as data is an Object and not a String at that moment, I would need to do data.toString() and that can launch an exception if null, so I will leave this here like it is.